icu4c/source/i18n/number_decimalquantity.cpp

```diff
@@ -881,12 +894,27 @@ UnicodeString DecimalQuantity::toPlainString() const {
     if (isNegative()) {
         sb.append(u'-');
     }
-    if (precision == 0 || getMagnitude() < 0) {
+    if (precision == 0) {
         sb.append(u'0');
+        return sb;
+    }
+    int upper = scale + precision + exponent - 1;
+    int lower = scale + exponent;
+    if (upper < lReqPos - 1) {
+        upper = lReqPos - 1;
+    }
+    if (lower > rReqPos) {
+        lower = rReqPos;
+    }    
+    int p = upper;
+    for (; p >= 0; p--) {
+        sb.append((char) ('0' + getDigitPos(p - scale - exponent)));
```

Comment:
sb is a UnicodeString, so you should append a 16-bit character.  `u'0'` and remove the `(char)` cast.

Reply:
Done.